File: grok/clients/node/test-connection.js
const { GrokClient } = require('./grok-client');

async function main() {
  console.log('\n🔌 Grok (xAI) Connection Test\n');
  const client = new GrokClient();
  let passed = 0, failed = 0;

  async function test(name, fn) {
    try { await fn(); console.log(`  ✅ ${name}`); passed++; }
    catch (e) { console.log(`  ❌ ${name}: ${e.message || JSON.stringify(e)}`); failed++; }
  }

  await test('Chat (grok-3-mini)', async () => {
    const res = await client.generateContent({
      prompt: 'Say "Grok connected" — nothing else.',
      model: 'grok-3-mini', max_tokens: 10
    });
    if (!res.text.toLowerCase().includes('grok') && !res.text.toLowerCase().includes('connected'))
      throw new Error(`Got: ${res.text}`);
  });

  await test('Streaming (grok-3-mini)', async () => {
    let text = '';
    for await (const chunk of client.generateContentStream({
      prompt: 'Count to 3.', model: 'grok-3-mini', max_tokens: 20
    })) { text += chunk.text; }
    if (!text) throw new Error('No output');
  });

  console.log(`\n  ${passed} passed · ${failed} failed`);
  process.exit(failed > 0 ? 1 : 0);
}

main().catch(e => { console.error(e); process.exit(1); });

File: grok/clients/python/test_connection.py
"""Test xAI Grok Python client connection"""
from grok_client import GrokClient


def main():
    print('\n🔌 Grok (xAI) Python Connection Test\n')
    client = GrokClient()
    passed = failed = 0

    def test(name, fn):
        nonlocal passed, failed
        try:
            fn()
            print(f'  ✅ {name}')
            passed += 1
        except Exception as e:
            print(f'  ❌ {name}: {e}')
            failed += 1

    def t_chat():
        res = client.generate_content(
            'Say "Grok connected" — nothing else.',
            model='grok-3-mini'
        )
        assert 'grok' in res['text'].lower() or 'connected' in res['text'].lower()

    def t_streaming():
        chunks = list(client.generate_content_stream(
            'Count to 3.', model='grok-3-mini', max_tokens=20
        ))
        text = ''.join(c['text'] for c in chunks if not c['done'])
        assert text, 'No streaming output'

    test('Chat completion (grok-3-mini)', t_chat)
    test('Streaming (grok-3-mini)', t_streaming)

    print(f'\n  {passed} passed · {failed} failed')
    exit(1 if failed else 0)


if __name__ == '__main__':
    main()
